- name: "mypy"
  cmd:
    - "mypy"
    - "--config-file"
    - "scripts/mypy.ini"
  extensions:
    - ".py"
  ignored_issues:
    - "error: \"Type[Flags]\" has no attribute"
    - "error: Source file found twice under different module names"
- name: "pylint"
  cmd:
    - "pylint"
    - "--rcfile"
    - "scripts/lint/pylintrc"
  extensions:
    - ".py"
- name: "pylint_permissive"
  cmd:
    - "pylint"
    - "--rcfile"
    - "scripts/lint/pylintrc_permissive"
  extensions:
    - ".py"
  run_by_default: false
- name: "cpplint"
  cmd:
    - "cpplint"
  extensions:
    - ".cc"
    - ".h"
  use_stderr: true
- name: "golint"
  cmd:
    - "golint"
  extensions:
    - ".go"
- name: "helmlint"
  cmd:
    - "python3"
    - "./scripts/lint/helm_lint.py"
  extensions:
    - ".yaml"
  use_stderr: true
